api/client/workflows/plugin.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workflows
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils"
+
+	"github.com/gravitational/trace"
+	"github.com/pborman/uuid"
+)
+
+// Plugin is a management plugin for Teleport access request.
+// It can be used to create, retrieve, or update requests.
+//
+// It can also be used to watch for request events from a event stream from the
+// Auth server. This can be useful for integrating with external tools.
+// We have already integrated with some popular external tools using this Plugin.
+// See https://goteleport.com/docs/enterprise/workflow/#integrating-with-an-external-tool.
+//
+// Plugin must have a client with an open connection to a Teleport Auth server in
+// order to function See pkg.go.dev/github.com/gravitational/teleport/api/client
+// for Teleport client documentation.
+type Plugin struct {
+	client *client.Client
+	name   string
+}
+
+// NewPlugin creates a new plugin using the given client and plugin name.
+func NewPlugin(ctx context.Context, name string, client *client.Client) *Plugin {
+	return &Plugin{
+		client: client,
+		name:   name,
+	}
+}
+
+// WatchRequests registers a new watcher for pending access requests.
+func (p *Plugin) WatchRequests(ctx context.Context, filter Filter) (RequestWatcher, error) {
+	return newRequestWatcher(ctx, p.client, filter)
+}
+
+// Close closes the plugin's underlying client.
+func (p *Plugin) Close() error {
+	return p.client.Close()
+}
```

Comment:
the caller passes the client in, so the caller should be the one closing the client
let's remove this method